lib/nbp/push_connector.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module Nbp
+  class SettingsError < StandardError; end
+  class ConnectorError < StandardError; end
+
+  class PushConnector
+    include Singleton
+
+    def initialize
+      super
+      create_source! unless source_exists?
+    end
+
+    def push_lom!(xml)
+      response = api_conn.put("/push-connector/api/lom-v2/#{settings.source.slug}") do |req|
+        req.body = {metadata: xml}.to_json
+      end
+      raise_connector_error('Could not push task LOM', response) unless success_status?(response.status)
+    end
+
+    def delete_task!(task_uuid)
+      response = api_conn.delete("/push-connector/api/course/#{settings.source.slug}/#{task_uuid}")
+      raise_connector_error('Could delete task', response) unless success_status?(response.status)
```

Comment:
We need to re-think `404` errors when deleting a task.

Let's image I created a task, which was saved successfully but couldn't be pushed due to another error. When I later delete this task again, it isn't found upstream, thus failing this request, too. However, the reported error 404 indicates that the task is (no longer) published online, so somewhat we can just proceed without any further error handling.